addons/website_sale_wishlist/models/product_wishlist.py

```diff
@@ -37,7 +37,19 @@ def current(self):
 
     @api.model
     def _add_to_wishlist(self, pricelist_id, currency_id, website_id, price, product_id, partner_id=False):
-        wish = self.env['product.wishlist'].create({
+        if partner_id:
+            wish = self.search([('partner_id', '=', partner_id), ('product_id', '=', product_id), '|', ('active', '=', False), ('active', '=', True)])
+            if wish:
+                wish.update({
+                    'active': True,
+                    'currency_id': currency_id,
+                    'pricelist_id': pricelist_id,
+                    'price': price,
+                    'website_id': website_id,
```

Comment:
I would create a dict with all the values and use the same dict in both update and create. Also why `update` and not `write` ?